src/main/java/io/cdap/plugin/gcp/gcs/sink/GCSBatchSink.java

```diff
@@ -137,12 +144,25 @@ private void emitMetrics(boolean succeeded, BatchSinkContext context) {
 
     try {
       StorageClient storageClient = StorageClient.create(config.getProject(), config.getServiceAccountFilePath());
-      storageClient.mapMetaDataForAllBlobs(outputPath, new MetricsEmitter(context.getMetrics())::emitMetrics);
+      storageClient.mapMetaDataForAllBlobs(getPrefixPath(), new MetricsEmitter(context.getMetrics())::emitMetrics);
     } catch (Exception e) {
       LOG.warn("Metrics for the number of affected rows in GCS Sink maybe incorrect.", e);
     }
   }
 
+  private String getPrefixPath() {
+    if (!config.getFileSystemProperties().containsKey(AVRO_NAMED_OUTPUT)) {
```

Comment:
this is the prefix property when the format is avro, but a different property is used for the other formats ("mapreduce.output.basename" from FileOutputFormat).

Also, I think it makes sense to add the file prefix as an explicit property in the advanced section along with the new FileSystemProperties. If the user sets a value, we can set it in the getFileSystemProperties() method. That will be a more user friendly way to do it, rather than forcing the user to know about these properties.

Reply:
Added the new property to take in file name prefix, this will have preference if the same properties are set using FileSystemProperties. Also considered the property for other formats.